File: server/src/models/Reel.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IReel extends Document {
  title: string;
  videoUrl: string;
  description?: string;
  displayOrder: number;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const ReelSchema = new Schema<IReel>(
  {
    title: { type: String, required: true },
    videoUrl: { type: String, required: true },
    description: { type: String },
    displayOrder: { type: Number, default: 0 },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

ReelSchema.index({ isActive: 1, displayOrder: 1 });

export const Reel =
  mongoose.models.Reel ||
  mongoose.model<IReel>('Reel', ReelSchema);

File: mobile/lib/models/reel.dart
class Reel {
  final String id;
  final String title;
  final String videoUrl;
  final String? description;
  final int displayOrder;
  final bool isActive;
  final DateTime? createdAt;

  Reel({
    required this.id,
    required this.title,
    required this.videoUrl,
    this.description,
    this.displayOrder = 0,
    this.isActive = true,
    this.createdAt,
  });

  factory Reel.fromJson(Map<String, dynamic> json) {
    return Reel(
      id: (json['_id'] ?? json['id'] ?? '').toString(),
      title: json['title'] ?? '',
      videoUrl: json['videoUrl'] ?? '',
      description: json['description'],
      displayOrder: json['displayOrder'] ?? 0,
      isActive: json['isActive'] ?? true,
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }
}
